File: frontend/src/services/notificationService.js
import apiClient from './apiClient';

export const notificationService = {
  getNotifications: async (email) => {
    const response = await apiClient.get(email ? `/api/notifications/${email}` : '/api/notifications');
    return response?.data?.data || [];
  },

  getUnreadNotifications: async (email) => {
    const response = await apiClient.get(email ? `/api/notifications/${email}/unread` : '/api/notifications/unread');
    return response?.data?.data || [];
  },

  getUnreadCount: async (email) => {
    const response = await apiClient.get(email ? `/api/notifications/${email}/unread/count` : '/api/notifications/unread/count');
    return response?.data?.data || 0;
  },

  markAsRead: async (notificationId) => {
    const response = await apiClient.put(`/api/notifications/${notificationId}/read`);
    return response?.data?.data;
  },

  markAllAsRead: async (email) => {
    const response = await apiClient.put(email ? `/api/notifications/${email}/read-all` : '/api/notifications/read-all');
    return response?.data?.data;
  },

  deleteNotification: async (notificationId) => {
    const response = await apiClient.delete(`/api/notifications/${notificationId}`);
    return response?.data?.data;
  },
};

File: backend/demo/src/main/java/com/example/demo/Service/NotificationService.java
package com.example.demo.Service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.example.demo.Entity.Notification;
import com.example.demo.Entity.User;
import com.example.demo.Repository.NotificationRepo;
import com.example.demo.dto.NotificationResponsedto;
import com.example.demo.mapper.NotificationMapper;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class NotificationService {
    
    private final NotificationRepo notificationRepo;
    
    public Notification createNotification(User recipient, String message, String type, 
                                          String relatedEntityType, Long relatedEntityId) {
        Notification notification = new Notification();
        notification.setRecipient(recipient);
        notification.setMessage(message);
        notification.setType(type);
        notification.setRelatedEntityType(relatedEntityType);
        notification.setRelatedEntityId(relatedEntityId);
        notification.setIsRead(false);
        return notificationRepo.save(notification);
    }
    
    public List<Notification> getNotifications(User user) {
        return notificationRepo.findByRecipientOrderByCreatedAtDesc(user);
    }
    
    public List<Notification> getUnreadNotifications(User user) {
        return notificationRepo.findByRecipientAndIsReadFalse(user);
    }
    
    public long getUnreadCount(User user) {
        return notificationRepo.countByRecipientAndIsReadFalse(user);
    }
    
    public Notification markAsRead(Long notificationId) {
        Notification notification = notificationRepo.findById(notificationId).orElse(null);
        if (notification != null) {
            notification.setIsRead(true);
            return notificationRepo.save(notification);
        }
        return null;
    }
    
    public void markAllAsRead(User user) {
        List<Notification> notifications = notificationRepo.findByRecipientAndIsReadFalse(user);
        notifications.forEach(n -> n.setIsRead(true));
        notificationRepo.saveAll(notifications);
    }
    
    public void deleteNotification(Long notificationId) {
        notificationRepo.deleteById(notificationId);
    }
}
